src/query_plan/operation.rs

```diff
@@ -782,6 +808,49 @@ pub(crate) mod normalized_field_selection {
         pub(crate) fn sibling_typename_mut(&mut self) -> &mut Option<Name> {
             &mut self.data.sibling_typename
         }
+
+        pub(crate) fn type_if_added_to(
+            &self,
+            parent_type: &CompositeTypeDefinitionPosition,
+            schema: &ValidFederationSchema,
+        ) -> Result<Option<CompositeTypeDefinitionPosition>, FederationError> {
+            let data = self.data();
+            if &data.field_position == parent_type {
+                let digest = match &self.data.field_position {
+                    FieldDefinitionPosition::Object(ty) => CompositeTypeDefinitionPosition::Object(
+                        ObjectTypeDefinitionPosition::new(ty.type_name.clone()),
+                    ),
+                    FieldDefinitionPosition::Interface(ty) => {
+                        CompositeTypeDefinitionPosition::Interface(
+                            InterfaceTypeDefinitionPosition::new(ty.type_name.clone()),
+                        )
+                    }
+                    FieldDefinitionPosition::Union(ty) => CompositeTypeDefinitionPosition::Union(
+                        UnionTypeDefinitionPosition::new(ty.type_name.clone()),
+                    ),
+                };
+                return Ok(Some(digest));
+            }
+            if data.name() == parent_type.type_name() {
+                todo!()
+            }
+            /*
+            if (this.name === typenameFieldName) {
+                return parentType.typenameField()?.type;
+            }
+            */
+            if self.can_rebase_on(parent_type, schema) {
+                // parent_type.field(self.data.name())?
+                todo!()
+            } else {
+                Ok(None)
+            }
+            /*
+            return this.canRebaseOn(parentType)
+                ? parentType.field(this.name)?.type
+                : undefined;
+            */
```

Comment:
Remaining question: We can the a `FieldDefinitionPosition`. Is it sound to construct a `CompositeTypeDefinitionPosition` with it (i.e. map the variants and remove the `field_name` from `FieldDefinitionPosition`?